tasks/push-snapshot/push-snapshot.yaml

```diff
@@ -41,15 +41,31 @@ spec:
         set -eux
 
         push_image () { # Expected arguments are [name, containerImage, repository, tag]
-          printf '* Pushing component: %s to %s:%s\n' "$1" "$3" "$4"
-          attempt=0
-          until [ "$attempt" -gt "$(params.retries)" ] ; do # 0 retries by default which will execute this once
-            cosign copy -f "$2" "$3:$4" && break
-            attempt=$((attempt+1))
-          done
-          if [ "$attempt" -gt "$(params.retries)" ] ; then
-            echo "Max retries exceeded."
-            exit 1
+          source_digest=$(skopeo inspect \
```

Comment:
How about moving this one back to the main script to only call it once per component? With just two floatingTags, you will have 4 skopeo inspect calls for the same source image where it could have been just 1.

So I'd suggest moving the skopeo inspect call to something like like 104 (in the `for component...` loop, before determining what tags we need) and then adding the resulting digest as another parameter of the `push_image` function.

The only downside I can see is that we'll have two instances of `skopeo inspect` in two different places, but that should be fine.

Reply:
Smart, updated

Reply:
Cool!